tests/integration/test_package_managers.py

```diff
@@ -147,52 +147,74 @@ def test_packages(
         test_params.repo, test_params.ref, f"{test_case}-source", tmpdir
     )
 
-    output_folder = os.path.join(tmpdir, f"{test_case}-output")
+    _ = utils.fetch_deps_and_check_output(
+        tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
+    )
+
+
+@pytest.mark.parametrize(
+    "test_params,check_cmd",
+    [
+        # Test case checks fetching retrodep dependencies, generating environment vars file,
+        # building image with all prepared prerequisites and printing help message for retrodep
+        # app in built image
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/retrodep.git",
+                ref="c3496edd5d45523a1ed300de1575a212b86d00d3",
+                packages=({"path": ".", "type": "gomod"},),
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+            ["retrodep", "--help"],
+            id="gomod_e2e_test",
+        ),
+    ],
+)
+def test_e2e(
+    test_params: utils.TestParameters,
+    check_cmd: List[str],
+    cachi2_image: utils.ContainerImage,
+    tmpdir: Path,
+    test_data_dir: Path,
+    request,
```

Comment:
Just a minor thing, but we can annotate this with `pytest.FixtureRequest`

Reply:
Done